### DISCUSSION TITLE
Esp32-S3 Firmware uploading Problem

### DISCUSSION THREAD

**Discussion by sheetaljain01** [micropython/micropython#16218] [General]
Hi, 

I have created custom firmware for the ESP32-S3 and obtained the following files: `micropython.bin`, `bootloader.bin`, and `partition-table.bin`.

While uploading the firmware using the following command:  

python3 -m esptool --chip esp32s3 --port COM5 write_flash 0x1000 bootloader.bin 0x8000 partition-table.bin 0x10000 micropython.bin

my ESP32-S3 keeps rebooting and shows this error continuously:  

ESP-ROM:esp32s3-20210327  
Build: Mar 27 2021  
rst:0x10 (RTCWDT_RTC_RST), boot:0x8 (SPI_FAST_FLASH_BOOT)  
SPIWP:0xee  
mode: DIO, clock div:1  
load: 0x3fce3808, len: 0x43c  
load: 0x403c9700, len: 0xbec  
load: 0x20736177, len: 0x6c6c6163  
Invalid image block, can't boot.  
ets_main.c 329  

This error keeps repeating. Can you help me troubleshoot the issue?"

**Comment by shariltumin** [comment]
You should have a firmware.bin after a build process. Then you can try:

```
python3 -m esptool --port COM5 --baud 460800 --before default_reset --after hard_reset --chip esp32s3  write_flash --flash_mode dio --flash_size detect --flash_freq 80m 0x0 firmware.bin

```

**Reply by sheetaljain01** [reply]
![Screenshot 2024-11-12 191933](https://github.com/user-attachments/assets/96a50feb-e21d-46e1-bbb2-29966536d399)
Thanks for your reply. I am compiling MicroPython with ADF support for the KORVO2V3 board. However, while building, I didn’t get the `firmware.bin` file in the build folder.  

I am following this repository: [https://github.com/espressif/esp-adf/tree/master/micropython_adf](https://github.com/espressif/esp-adf/tree/master/micropython_adf).  

Could you help me resolve this issue?

**Reply by shariltumin** [reply]
Then please try:

```
python3 esptool --port COM5 --baud 460800 --before default_reset --after no_reset --chip esp32s3  write_flash --flash_mode dio --flash_size detect --flash_freq 80m 0x0 bootloader.bin 0x8000 partition-table.bin 0x10000 micropython.bin
```

**Reply by sheetaljain01** [reply]
Thanks [shariltumin](https://github.com/shariltumin) it work

**Reply by shariltumin** [reply]
Nice!

**Comment by sosi-deadeye** [comment]
The offset of the firmware is `0x0` and not `0x1000`. 

| µController | Offset |
| --- | --- |
| ESP32 | `0x1000` |
| ESP32S2 | `0x1000` |
| ESP32S3 | `0x0` |
| ESP32C3 | `0x0` |
| ESP32C6 | `0x0` |

**Reply by sheetaljain01** [reply]
Thanks [sosi-deadeye](https://github.com/sosi-deadeye)
